src/store/resumable-task-store-level.ts

```diff
@@ -0,0 +1,119 @@
+import type { ManagedResumableTask, ResumableTaskStore } from '../types/resumable-task-store.js';
+
+import { Cid } from '../utils/cid.js';
+import { createLevelDatabase, LevelWrapper } from './level-wrapper.js';
+
+type ResumableTaskStoreLevelConfig = {
+  location?: string,
+  createLevelDatabase?: typeof createLevelDatabase
+};
+
+/**
+ * A simple single-instance implementation of {@link ResumableTaskStore} that works in both browsers and node.js.
+ * Leverages LevelDB under the hood.
+ */
+export class ResumableTaskStoreLevel implements ResumableTaskStore {
+  private static readonly taskTimeoutInSeconds = 60;
+
+  db: LevelWrapper<string>;
+  config: ResumableTaskStoreLevelConfig;
+
+  constructor(config: ResumableTaskStoreLevelConfig) {
+    this.config = {
+      // defaults:
+      location: 'RESUMABLE-TASK-STORE',
+      createLevelDatabase,
+      // user-provided overrides:
+      ...config,
+    };
+
+    this.db = new LevelWrapper<string>({
+      location            : this.config.location!,
+      createLevelDatabase : this.config.createLevelDatabase,
+      keyEncoding         : 'utf8'
+    });
+  }
+
+  async open(): Promise<void> {
+    await this.db.open();
+  }
+
+  async close(): Promise<void> {
+    await this.db.close();
+  }
+
+  public async register(task: any, timeoutInSeconds: number): Promise<ManagedResumableTask> {
+    const taskId = await Cid.computeCid(task);
+
+    const managedResumableTask: ManagedResumableTask = {
+      id         : taskId,
+      timeout    : Date.now() + (timeoutInSeconds * 1000),
+      retryCount : 0,
+      task,
+    };
+
+    await this.db.put(taskId, JSON.stringify(managedResumableTask));
+
+    return managedResumableTask;
+  }
+
+  public async grab(count: number): Promise<ManagedResumableTask[]> {
+    const tasks: ManagedResumableTask[] = [];
+
+    // iterate over the tasks to find unhandled tasks to return to the caller
```

Comment:
Sanity: If I understand the logic correctly, maybe we should add a note here referencing that we grab tasks that have exceeded their timeout as the ones who haven't might already be running.

Also worth mentioning here that `delete()` is called within `runWithAutomaticTimeoutExtension` to remove the task from the db after the task has completed.

Reply:
Done. Let me know if I need to add or change more!